internal/aws/proxy/server.go

```diff
@@ -101,7 +100,7 @@ func NewServer(cfg *Config, logger *zap.Logger) (Server, error) {
 			}
 
 			// Sign request. signer.Sign() also repopulates the request body.
-			_, err = signer.Sign(req, body, service, *awsCfg.Region, time.Now())
+			_, err = signer.Sign(req, body, cfg.ServiceName, *awsCfg.Region, time.Now())
```

Comment:
any consequence if `ServiceName` is empty or consists of an invalid service such as "foo"?

Reply:
so, I just ran the new code leaving `service_name` as empty quotes and OTEL starts just fine:
```
$ curl http://localhost:2000/                                                                                                     
{"message":"Credential should be scoped to correct service: 'es'. "}
```

Reply:
I'm testing it with an OpenSearch endpoint. Hence, my AWS is telling me to put "es" as service name which is expected.

Reply:
I also tried with `foo` and get same response. So, I can safely say, there are no consequences

Reply:
Well, sure. It fails gracefully with an error message that makes sense, so there's that.